llvm/utils/update_analyze_test_checks.py

```diff
@@ -61,163 +215,17 @@ def main():
-
-            builder.processed_prefixes(prefixes)
-
-        func_dict = builder.finish_and_get_func_dict()
-        is_in_function = False
-        is_in_function_start = False
-        prefix_set = set([prefix for prefixes, _ in prefix_list for prefix in prefixes])
-        common.debug("Rewriting FileCheck prefixes:", str(prefix_set), file=sys.stderr)
-        output_lines = []
-
-        generated_prefixes = []
-        for input_info in ti.iterlines(output_lines):
-            input_line = input_info.line
-            args = input_info.args
-            if is_in_function_start:
-                if input_line == "":
-                    continue
-                if input_line.lstrip().startswith(";"):
-                    m = common.CHECK_RE.match(input_line)
-                    if not m or m.group(1) not in prefix_set:
-                        output_lines.append(input_line)
-                        continue
-
-                # Print out the various check lines here.
-                generated_prefixes.extend(
-                    common.add_analyze_checks(
-                        output_lines,
-                        ";",
-                        prefix_list,
-                        func_dict,
-                        func_name,
-                        ginfo,
-                        is_filtered=builder.is_filtered(),
-                    )
-                )
-                is_in_function_start = False
-
-            if is_in_function:
-                if common.should_add_line_to_output(input_line, prefix_set):
-                    # This input line of the function body will go as-is into the output.
-                    output_lines.append(input_line)
-                else:
-                    continue
-                if input_line.strip() == "}":
-                    is_in_function = False
-                continue
-
-            # If it's outside a function, it just gets copied to the output.
-            output_lines.append(input_line)
-
-            m = common.IR_FUNCTION_RE.match(input_line)
-            if not m:
-                continue
-            func_name = m.group(1)
-            if ti.args.function is not None and func_name != ti.args.function:
-                # When filtering on a specific function, skip all others.
-                continue
-            is_in_function = is_in_function_start = True
-
-        if ti.args.gen_unused_prefix_body:
-            output_lines.extend(
-                ti.get_checks_for_unused_prefixes(prefix_list, generated_prefixes)
-            )
-
-        common.debug("Writing %d lines to %s..." % (len(output_lines), ti.path))
-
-        with open(ti.path, "wb") as f:
-            f.writelines(["{}\n".format(l).encode("utf-8") for l in output_lines])
+        try:
+            update_test(opt_basename, ti)
+        except Exception:
+            stderr.write(f"Error: Failed to update test {ti.path}\n")
+            print_exc()
+            returncode = 1
+    return returncode
 
 
```

Comment:
I can't comment on the context, but should this be `sys.exit(main())` like the rest, or is this a special case?